components/toggle_modal_button_redux/toggle_modal_button_redux.jsx

```diff
@@ -81,6 +91,7 @@ class ModalToggleButtonRedux extends React.PureComponent {
                 onClick={clickHandler}
             >
                 {children}
+                {badge}
```

Comment:
Instead of adding an extra prop to ToggleModalButtonRedux that's specific to this ccase, couldn't the MenuItemToggleModalReduxImpl (now that's a mouthful) pass the badge in as one of this component's children?

Reply:
Yes, that works better and cleans-up unneeded parameter passing. I've done the changes but will not check them in right now to not derail the QA'd status. I've opened https://mattermost.atlassian.net/browse/MM-34438 to check it in as a follow-up.